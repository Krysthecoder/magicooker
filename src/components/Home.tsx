import { HomeProps } from './types';

const Home: React.FC<HomeProps> = ({ title }) => {
  return (
    <div>
      <h1>{title}</h1>
    </div>
  );
};

export default Home;
