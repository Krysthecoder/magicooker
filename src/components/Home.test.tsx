import { render, screen } from '@testing-library/react';
import Home from './Home';

describe('Hello Component', () => {
  it('renders with title << Hello world >>', () => {
    render(<Home title="Hello world" />);
    const testText = screen.getByText('Hello world');
    expect(testText).toBeInTheDocument();
  });

  it('renders with title << MagiCooker Recipes >>', () => {
    render(<Home title="MagiCooker Recipes," />);
    const homeTitle = screen.getByText('MagiCooker Recipes,');
    expect(homeTitle).toBeInTheDocument();
  });

  it('renders correctly with an empty title', () => {
    render(<Home title="" />);
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('');
  });

  it('renders correctly with an undefined title', () => {
    render(<Home title={undefined} />);
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Home title="Snapshot test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an h1 element for accessibility', () => {
    render(<Home title="Accessibility check" />);
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe('H1');
  });
});
