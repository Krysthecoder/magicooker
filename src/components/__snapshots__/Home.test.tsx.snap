// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Snapshot test
    </h1>
  </div>
</DocumentFragment>
`;
